### PR TITLE
Admin login

### PR DESCRIPTION
**Pull request by EverardoAlcarazBaltazar**
# Description

We implement a login form and database for admins, and also linked them because is necessary that only the admins can manage the employee and store table.

## Screenshots

login
Captura

### CODE REVIEW

**Review by DiegoCoronaBarragan — APPROVED** [review]

**Review by oscaregr — APPROVED** [review]

File: app/assets/stylesheets/layout/_navbar.css.scss [hunk 1]
```diff
@@ -1,32 +1,40 @@
 @import '../abstracts/variables.css';
 
 nav{
-    display: flex;
-    justify-content: space-between;
-    background-color: #D9D9D9;    ;
     align-items: center;
-    height: 72px;
+    background-color: #D9D9D9;    ;
```
Comment (JuanequeX) [inline app/assets/stylesheets/layout/_navbar.css.scss:5]: Changes for a color variable

File: app/assets/stylesheets/pages/_report_form.css.scss [hunk 2]
```diff
@@ -23,5 +25,21 @@
...
+      height: 26px;
+      left: 495px;
+      padding-left: 5px;
+      top: 390px;
+      width: 264px;
+    }
+    input:hover{
+      background-color: $placeholder-bg-color-hover;
+    }
+}
```
Comment (JuanequeX) [inline app/assets/stylesheets/pages/_report_form.css.scss:45]: add an space between your class name an your curly braces, and add a blank line between each blocks

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
```
Comment (JuanequeX) [inline app/models/user.rb:3]: Delete this comments

File: app/views/users/confirmations/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,16 @@
+<h2>Resend confirmation instructions</h2>
+
+<%= form_for(resource, as: resource_name, url: confirmation_path(resource_name), html: { method: :post }) do |f| %>
+  <%= render "users/shared/error_messages", resource: resource %>
+
+  <div class="field">
+    <%= f.label :email %><br />
+    <%= f.email_field :email, autofocus: true, autocomplete: "email", value: (resource.pending_reconfirmation? ? resource.unconfirmed_email : resource.email) %>
```
Comment (JuanequeX) [inline app/views/users/confirmations/new.html.erb:8]: This line is so long, please refactor it.

File: app/views/users/mailer/confirmation_instructions.html.erb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+<p>Welcome <%= @email %>!</p>
+
+<p>You can confirm your account email through the link below:</p>
+
+<p><%= link_to 'Confirm my account', confirmation_url(@resource, confirmation_token: @token) %></p>
```
Comment (JuanequeX) [inline app/views/users/mailer/confirmation_instructions.html.erb:5]: I think the tag `p` is unnecessary

File: app/views/users/mailer/unlock_instructions.html.erb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+<p>Hello <%= @resource.email %>!</p>
+
+<p>Your account has been locked due to an excessive number of unsuccessful sign in attempts.</p>
+
+<p>Click the link below to unlock your account:</p>
+
+<p><%= link_to 'Unlock my account', unlock_url(@resource, unlock_token: @token) %></p>
```
Comment (JuanequeX) [inline app/views/users/mailer/unlock_instructions.html.erb:7]: I think the tag p is unnecessary

File: app/views/users/mailer/reset_password_instructions.html.erb [hunk 7]
```diff
@@ -0,0 +1,8 @@
+<p>Hello <%= @resource.email %>!</p>
+
+<p>Someone has requested a link to change your password. You can do this through the link below.</p>
+
+<p><%= link_to 'Change my password', edit_password_url(@resource, reset_password_token: @token) %></p>
```
Comment (JuanequeX) [inline app/views/users/mailer/reset_password_instructions.html.erb:5]: I think the tag p is unnecessary

File: app/views/users/registrations/edit.html.erb [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+  <div class="field">
+    <%= f.label :current_password %> <i>(we need your current password to confirm your changes)</i><br />
+    <%= f.password_field :current_password, autocomplete: "current-password" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Update" %>
+  </div>
+<% end %>
```
Comment (JuanequeX) [inline app/views/users/registrations/edit.html.erb:37]: the use of the `<br>`is a bad practice, please remove it

File: app/views/users/registrations/new.html.erb [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+  <div class="field">
+    <%= f.label :email %><br />
+    <%= f.email_field :email, autofocus: true, autocomplete: "email" %>
+  </div>
+
+  <div class="field">
+    <%= f.label :password %>
+    <% if @minimum_password_length %>
+    <em>(<%= @minimum_password_length %> characters minimum)</em>
+    <% end %><br />
```
Comment (JuanequeX) [inline app/views/users/registrations/new.html.erb:15]: Same here with the `<br>`

File: app/views/users/sessions/new.html.erb [hunk 10]
```diff
@@ -0,0 +1,23 @@
+<div class="main-container">
+  <div class="login-container">
+    <section class="title-container">
+      <h1>Log-in</h1>
+    </section>
+    <p class="navbar-text float-right"></p>
```
Comment (JuanequeX) [inline app/views/users/sessions/new.html.erb:6]: what is the reason of put this `p`?

File: db/seeds.rb [hunk 11]
```diff
@@ -43,3 +43,6 @@
   end
   employee_number = 0
 end
+
+User.create!({:email => "admin@gmail.com", :password => "123456", :password_confirmation => "123456" })
+User.create!({:email => "iker@gmail.com", :password => "123456", :password_confirmation => "123456" })
```
Comment (JuanequeX) [inline db/seeds.rb:48]: Add a blank line at the end of this file

**Review by JuanequeX — CHANGES_REQUESTED** [review]

**Review by MarcoRodriguez23 — APPROVED** [review]

File: app/assets/stylesheets/layout/_navbar.css.scss [hunk 12]
```diff
@@ -1,40 +1,42 @@
...
-    }
-    .logo:hover{
-      opacity: 0.6;
-    }
+  .logo:hover {
+    opacity: 0.6;
+  }
+
+  .nav-button {
+    background-color: #7B7B7B;
```
Comment (JuanequeX) [inline app/assets/stylesheets/layout/_navbar.css.scss:24]: Changes for a color variable

File: app/assets/stylesheets/layout/_navbar.css.scss [hunk 13]
```diff
@@ -1,40 +1,42 @@
...
+    border: none;
+    color: white;
```
Comment (JuanequeX) [inline app/assets/stylesheets/layout/_navbar.css.scss:26]: Changes for a color variable

File: app/assets/stylesheets/layout/_navbar.css.scss [hunk 14]
```diff
@@ -1,40 +1,42 @@
...
-        width: 168px;
-      }
-    .nav-button:hover{
-      background-color: $placeholder-bg-color-hover;
-    }
-}
+  .nav-button:hover {
+    background-color: $placeholder-bg-color-hover;
+  }
+}
```
Comment (JuanequeX) [inline app/assets/stylesheets/layout/_navbar.css.scss:42]: Add a blank line at the end of this file

File: app/assets/stylesheets/pages/_checking_employee.css.scss [hunk 15]
```diff
@@ -109,10 +112,11 @@
...
         }
+
         button:hover {
           box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
         }
       }
     }
   }
-}
+}
```
Comment (JuanequeX) [inline app/assets/stylesheets/pages/_checking_employee.css.scss:122]: Add a blank line at the end of this file

File: app/assets/stylesheets/pages/_crud_employee.css.scss [hunk 16]
```diff
@@ -187,4 +187,4 @@
       }
     }
   }
-}
+}
```
Comment (JuanequeX) [inline app/assets/stylesheets/pages/_crud_employee.css.scss:190]: Add a blank line at the end of this file

File: app/assets/stylesheets/pages/_crud_store.css.scss [hunk 17]
```diff
@@ -170,4 +171,4 @@ a{
       }
     }
   }
-}
+}
```
Comment (JuanequeX) [inline app/assets/stylesheets/pages/_crud_store.css.scss:174]: Add a blank line at the end of this file

File: app/assets/stylesheets/pages/_login_admin.css.scss [hunk 18]
```diff
@@ -104,4 +106,4 @@
       }
     }
   }
-}
+}
```
Comment (JuanequeX) [inline app/assets/stylesheets/pages/_login_admin.css.scss:109]: Add a blank line at the end of this file

File: app/assets/stylesheets/pages/_register_forms.css.scss [hunk 19]
```diff
@@ -109,4 +110,4 @@
       }
     }
   }
-}
+}
```
Comment (JuanequeX) [inline app/assets/stylesheets/pages/_register_forms.css.scss:113]: Add a blank line at the end of this file

File: app/assets/stylesheets/pages/_report_form.css.scss [hunk 20]
```diff
@@ -1,45 +1,48 @@
...
+    left: 495px;
+    padding-left: 5px;
+    top: 390px;
+    width: 264px;
+  }
+
+  input:hover {
+    background-color: $placeholder-bg-color-hover;
+  }
+}
```
Comment (JuanequeX) [inline app/assets/stylesheets/pages/_report_form.css.scss:48]: Add a blank line at the end of this file

**Review by JuanequeX — CHANGES_REQUESTED** [review]

**Review by JuanequeX — APPROVED** [review]

**Pull request merged**